OpenRA.Mods.Common/Traits/Tooltip.cs

```diff
@@ -54,18 +54,20 @@ public class TooltipInfo : TooltipInfoBase, ITooltipInfo
 		public string TooltipForPlayerStance(PlayerRelationship stance)
 		{
 			if (stance == PlayerRelationship.None || !GenericVisibility.HasStance(stance))
-				return Name;
+				return Game.Translation.Get(Name);
```

Comment:
What is the benefit and what should happen to `Map.Translation.Get(text)` then?

Reply:
The benefit is that it hides the implementation details from the callers, which makes the code more maintainable by allowing us to modify (or even replace) the backend without having to potententially change everything that uses it.

Maps should expose its own `Map.Translate(key, args)` method that checks the local map translations before falling back to Game. It would be better if this could whatever mechanism Fluent has for combining/overriding multiple translation files into a single context than to actually call `Game.Translate`. If this requires access to the underlying `Translation` object then it would be ok to declare that as `internal` instead of private to limit its use outside of Game and Map.

Anything related to traits should then be using `Map.Translate`, including lobby checkboxes and production tooltips. Anything that is UI related should be using `Game.Translate` to make sure that maps can't modify things that they shouldn't.

Reply:
> whatever mechanism Fluent has for combining/overriding multiple translation files
none / first hit is the translation

Reply:
I managed to hide the translation in wrapper function. However, lobby checkboxes don't have access to the map at the stage where they are set up and there is no merging of translations (yet).